core/src/main/java/org/elasticsearch/ingest/core/Pipeline.java

```diff
@@ -51,7 +51,11 @@ public Pipeline(String id, String description, CompoundProcessor compoundProcess
      * Modifies the data of a document to be indexed based on the processor this pipeline holds
      */
     public void execute(IngestDocument ingestDocument) throws Exception {
-        compoundProcessor.execute(ingestDocument);
+        try {
+            compoundProcessor.execute(ingestDocument);
+        } catch (CompoundProcessorException e) {
+            throw new IllegalArgumentException(e.getMessage());
```

Comment:
This was updated so that we do not expose `CompoundProcessorException` outside of the Ingest Pipeline.